src/main/java/com/sivalabs/springblog/post/domain/PostEntity.java

```diff
@@ -0,0 +1,183 @@
+package com.sivalabs.springblog.post.domain;
+
+import com.sivalabs.springblog.user.domain.UserEntity;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.SequenceGenerator;
+import jakarta.persistence.Table;
+import java.time.LocalDateTime;
+
+@Entity
+@Table(name = "posts")
+public class PostEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "post_id_seq_gen")
+    @SequenceGenerator(name = "post_id_seq_gen", sequenceName = "post_id_seq", allocationSize = 50)
+    private Long id;
+
+    @Column(nullable = false, unique = true, length = 255)
+    private String title;
+
+    @Column(nullable = false, unique = true, length = 255)
+    private String slug;
+
+    @Column(name = "short_description", nullable = false, length = 255)
+    private String shortDescription;
+
+    @Column(name = "content_markdown", nullable = false, columnDefinition = "TEXT")
+    private String contentMarkdown;
+
+    @Column(name = "content_html", nullable = false, columnDefinition = "TEXT")
+    private String contentHtml;
+
+    @Column(nullable = false, length = 20)
+    @Enumerated(EnumType.STRING)
+    private Status status;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "created_by", nullable = false)
+    private UserEntity createdBy;
+
+    @Column(name = "created_date", nullable = false, updatable = false)
+    private LocalDateTime createdDate;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "updated_by")
+    private UserEntity updatedBy;
+
+    @Column(name = "updated_date")
+    private LocalDateTime updatedDate;
+
+    // Enum for Status
+    public enum Status {
+        DRAFT,
+        PUBLISHED
+    }
+
+    // Default constructor
+    public PostEntity() {}
+
+    // Constructor with fields
+    public PostEntity(
+            Long id,
+            String title,
+            String slug,
+            String shortDescription,
+            String contentMarkdown,
+            String contentHtml,
+            Status status,
+            UserEntity createdBy,
+            LocalDateTime createdDate,
+            UserEntity updatedBy,
+            LocalDateTime updatedDate) {
+        this.id = id;
+        this.title = title;
+        this.slug = slug;
+        this.shortDescription = shortDescription;
+        this.contentMarkdown = contentMarkdown;
+        this.contentHtml = contentHtml;
+        this.status = status;
+        this.createdBy = createdBy;
+        this.createdDate = createdDate;
+        this.updatedBy = updatedBy;
+        this.updatedDate = updatedDate;
+    }
+
+    // Getters and Setters
```

Comment:
@geovannymcode can we use lombok annotations to remove boilerplate code? For example as below

@Getter,@Setter,@AllArgsConstructor,@NoArgsConstructor

cc @sivaprasadreddy

Reply:
Thanks for the suggestion! While Lombok is great for reducing boilerplate, I prefer to avoid it in this project for several reasons:

1. It can sometimes introduce compatibility issues with IDEs, build tools, or static analysis tools.
2. Having explicit getters, setters, and constructors makes the code easier to read and debug for new developers or during reviews.
3. This approach avoids adding an additional dependency and provides full control over the generated methods, which may be useful for future customizations or validations.

Let me know if there’s a specific case where using Lombok would add significant value here!